### PR TITLE
implement BUDDHA and IQDQD after death

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

File: Source/p_inter.c [hunk 1]
```diff
@@ -837,6 +837,11 @@ void P_DamageMobj(mobj_t *target,mobj_t *inflictor, mobj_t *source, int damage)
         }
 
       player->health -= damage;       // mirror mobj health here for Dave
+      // BUDDHA cheat
```
Comment (fabiangreffrath) [inline Source/p_inter.c:840]: So, the BUDDHA cheat prevents the player from dying. Am I wrong, I think I remembered that this cheat would even slowly recover the cheating player's health over time?

File: Source/d_player.h [hunk 2]
```diff
@@ -74,7 +74,9 @@ typedef enum
   // Not really a cheat, just a debug aid.
   CF_NOMOMENTUM       = 4,
   // [FG] FPS counter widget
-  CF_SHOWFPS          = 8
+  CF_SHOWFPS          = 8,
```
Comment (fabiangreffrath) [inline Source/d_player.h:77]: SHOWFPS could as well get moved to the `powers[]` array. What do you think?

File: Source/p_inter.c (hunk 1 above)
Comment (rfomin) [reply Source/p_inter.c:840]: No, I checked in GZDoom and health is not recovering. This cheat is about health management.

File: Source/d_player.h (hunk 2 above)
Comment (rfomin) [reply Source/d_player.h:77]: Ok, should I do it in this PT?

File: Source/m_cheat.c [hunk 3]
```diff
@@ -354,6 +358,25 @@ static void cheat_choppers()
...
+  if (plyr->playerstate == PST_DEAD)
+  {
+    signed int an;
+    mapthing_t mt = {0};
+    extern void P_SpawnPlayer (mapthing_t* mthing);
+
+    mt.x = plyr->mo->x >> FRACBITS;
+    mt.y = plyr->mo->y >> FRACBITS;
+    mt.angle = (plyr->mo->angle + ANG45/2)*(uint64_t)45/ANG45;
+    mt.type = consoleplayer + 1;
```
[the remark this answers is not in the document]
Comment (fabiangreffrath) [reply Source/m_cheat.c:371]: Ok, seems this was a PrBoom'ism and isn't actually required in MBF:
https://github.com/coelckers/prboom-plus/blob/db29f6f5dd760af82328bb945c738a04b6726ccb/prboom2/src/p_mobj.c#L1159

File: Source/d_player.h (hunk 2 above)
Comment (fabiangreffrath) [reply Source/d_player.h:77]: Ok.

Comment (rfomin) [reply Source/d_player.h]: Done in https://github.com/fabiangreffrath/woof/pull/448/commits/b3eaf9488bf57240dd9487f8be7ad57e1abef863

File: Source/doomdef.h [hunk 4]
```diff
@@ -215,6 +215,7 @@ typedef enum {
   NUMPOWERS,
   pw_mapcoords,
   pw_renderstats,
+  pw_fps
```
Comment (fabiangreffrath) [inline Source/doomdef.h:218]: Sorry for being picky, but could you please call it `pw_showfps` instead?

**Pull request merged**